src/fidesops/ops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -483,7 +485,19 @@ def _filter_privacy_request_queryset(
             PrivacyRequest.finished_processing_at > errored_gt,
         )
 
-    return query.order_by(PrivacyRequest.created_at.desc())
+    return query
+
+
+def _sort_privacy_request_queryset(
+    query: Query, sort_due_date: Optional[ColumnSort]
+) -> Query:
+    if sort_due_date is None:
+        return query.order_by(PrivacyRequest.created_at.desc())
+
+    if sort_due_date is ColumnSort.ASC:
+        return query.order_by(nullslast(PrivacyRequest.due_date.asc()))
+
+    return query.order_by(nullslast(PrivacyRequest.due_date.desc()))
```

Comment:
This function might be better off as:

```
def _sort_privacy_request_queryset(
    query: Query,
    sort_field: str = "created_at",
    sort_direction: str = "desc",
) -> Query:
    sort_object_attribute = getattr(PrivacyRequest, sort_field)
    sort_func = getattr(sort_object_attribute, sort_direction)
    return query.order_by(nullslast(sort_func()))
```

We could add some error handling or Enum support to mitigate the brittle nature of the string matching, and then we have a sort function we can extend to any field and direction.

Reply:
Great idea! I implemented this and added an additional validation that checks if the model has the request `sort_field`. The server returns a 500 error otherwise when trying to sort on a non existent field.